src/main/java/edu/ucsb/cs156/happiercows/controllers/CommonsController.java

```diff
@@ -125,6 +125,19 @@ public Commons getCommonsById(
     return ResponseEntity.ok().body(body);
   }
 
+  @ApiOperation(value = "Delete a Commons")
+  @PreAuthorize("hasRole('ROLE_ADMIN')")
+  @DeleteMapping("")
+  public ResponseEntity<String> deleteCommons(
+          @ApiParam("id") @RequestParam Long id) {
+      
+      Commons foundCommons = commonsRepository.findById(id).orElseThrow( ()->new EntityNotFoundException(Commons.class, id));
+ 
+      commonsRepository.deleteById(id);
+      return ResponseEntity.ok().body(String.format("commons with id %d deleted", id));
```

Comment:
API controller methods should ideally return JSON responses at all times. There is a `ApiController#genericMessage` method to help you generate that. If you return the object generated by that method in your controller, Spring Boot will format it correctly as JSON.

Reply:
solved in latest commit.